tests/test_packages/test_skills/test_coin_price/test_handlers.py

```diff
@@ -0,0 +1,240 @@
+"""This module contains the tests of the handler classes of the simple_data_request skill."""
+
+import logging
+from pathlib import Path
+from typing import cast
+from unittest.mock import patch
+
+from aea.protocols.dialogue.base import DialogueMessage
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.protocols.http.message import HttpMessage
+from packages.fetchai.skills.coin_price.dialogues import HttpDialogues
+from packages.fetchai.skills.coin_price.handlers import HttpHandler
+from packages.fetchai.skills.coin_price.models import CoinPriceModel
+
+from tests.conftest import ROOT_DIR
+
+
+class TestHttpHandler(BaseSkillTestCase):
+    """Test http handler of coin_price skill."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "coin_price")
+
+    @classmethod
+    def setup(cls, **kwargs):
+        """Setup the test class."""
+        super().setup()
+        cls.http_handler = cast(HttpHandler, cls._skill.skill_context.handlers.http)
+        cls.logger = cls._skill.skill_context.logger
+
+        cls.coin_price_model = cast(
+            CoinPriceModel, cls._skill.skill_context.coin_price_model
+        )
+
+        cls.http_dialogues = cast(
+            HttpDialogues, cls._skill.skill_context.http_dialogues
+        )
+
+        cls.list_of_messages = (
+            DialogueMessage(
+                HttpMessage.Performative.REQUEST,
+                {
+                    "method": "get",
+                    "url": "some_url",
+                    "headers": "",
+                    "version": "",
+                    "body": b"",
+                },
+            ),
+        )
+
+    def test_setup(self):
+        """Test the setup method of the http handler."""
+        assert self.http_handler.setup() is None
+        self.assert_quantity_in_outbox(0)
+
+    def test_handle_response(self):
+        """Test the _handle_response method of the http handler to a valid coin price response."""
+        # setup
+        http_dialogue = self.prepare_skill_dialogue(
+            dialogues=self.http_dialogues, messages=self.list_of_messages[:1],
+        )
+        incoming_message = self.build_incoming_message_for_skill_dialogue(
+            dialogue=http_dialogue,
+            performative=HttpMessage.Performative.RESPONSE,
+            version="",
+            status_code=200,
+            status_text="",
+            headers="",
+            body=b'{"fetch-ai":{"usd":100.00}}',
+        )
+
+        # handle message
+        self.http_handler.handle(incoming_message)
+
+        # check that data was correctly entered into shared state
+        oracle_data = {"value": 10000000, "decimals": self.coin_price_model.decimals}
+        assert self.http_handler.context.shared_state["oracle_data"] == oracle_data
+
+        # check that outbox contains update_prometheus metric message
+        self.assert_quantity_in_outbox(1)
```

Comment:
You have many occurrence of just checking the number of messages in the outbox without checking the message attributes. Make sure in all these case, there isn't an attribute that is key for the message to have that haven't checked for.